paimon-core/src/main/java/org/apache/paimon/tag/TagTimeExpire.java

```diff
@@ -52,24 +59,41 @@ private TagTimeExpire(
         this.callbacks = callbacks;
     }
 
-    public void run() {
+    public List<String> expire() {
         List<Pair<Tag, String>> tags = tagManager.tagObjects();
+        FileIO fileIO = snapshotManager.fileIO();
+        List<String> expired = new ArrayList<>();
         for (Pair<Tag, String> pair : tags) {
             Tag tag = pair.getLeft();
             String tagName = pair.getRight();
             LocalDateTime createTime = tag.getTagCreateTime();
-            Duration timeRetained = tag.getTagTimeRetained();
-            if (createTime == null || timeRetained == null) {
-                continue;
+            if (createTime == null) {
+                FileStatus tagFileStatus;
+                try {
+                    tagFileStatus = fileIO.getFileStatus(tagManager.tagPath(tagName));
```

Comment:
if (timeRetained != null && olderThanTime != null), in this case, we get file status for this tag.

Reply:
maybe `if (createTime == null && olderThanTime != null)` ?
if timeRetained != null, the tag has own createTime, no need to get file status.